metrics/line-numbers/fineLineNumber.go

```diff
@@ -0,0 +1,48 @@
+package linenumbers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strconv"
+
+	"gopkg.in/yaml.v3"
+)
+
+func recursiveFindLine(node *yaml.Node, keyIndex int, maxDepth int, keys []string) (*yaml.Node, error) {
+	for i := 0; i < len(node.Content); i += 2 {
+		key := node.Content[i]
+		val := node.Content[i+1]
+		if keyIndex+1 == maxDepth && val.Value == keys[maxDepth] {
+			return val, nil
+		}
+		if key.Value == keys[keyIndex] {
+			switch val.Kind {
+			case yaml.SequenceNode:
+				nextKeyIndex, err := strconv.Atoi(keys[keyIndex+1])
+				if err != nil {
+					return nil, err
+				}
+				return recursiveFindLine(val.Content[nextKeyIndex], keyIndex+2, maxDepth, keys)
+			default:
+				return recursiveFindLine(val, keyIndex+1, maxDepth, keys)
+			}
+		} else {
+			continue
+		}
+
+	}
+	return &yaml.Node{}, nil
+}
+
+// FindYamlLine returns a line number for a field within a yaml file given the path and yaml file.
+func FindYamlLine(filename string, keys []string, token string) (*yaml.Node, error) {
```

Comment:
Let's call this "FindNode" - this public interface would not conflict with the private name of the function (above) and the name similarity would help make their relationship clear.